test/test_trixi.jl

```diff
@@ -128,3 +128,40 @@ macro test_nowarn_mod(expr)
     end
   end
 end
+
+
+"""
+    @trixi_testset "name of the testset" #= code to test #=
+
+Similar to `@testset`, but wraps the code inside a temporary module to avoid
+namespace pollution. It also `include`s this file again to provide the
+definition of `@test_trixi_include`.
+"""
+macro trixi_testset(name, expr)
+  @assert name isa String
+  # TODO: `@eval` is evil
+  # We would like to use
+  #   mod = gensym(name)
+  # to create new module names for every test set. However, this is not
+  # compatible with the dirty hack using `@eval` to get the mapping when
+  # loading structured, curvilinear meshes. Thus, we need to use a plain
+  # module name here.
+  mod = Symbol("TrixiTestModule")
+  quote
+    @eval module $mod
+      using Test
+      using Trixi
+      include(@__FILE__)
```

Comment:
But doesn't that mean we have to be `using Plots` etc. always inside the `@trixi_testset` block? Or why are `Test` and `Trixi` special cases here?

Reply:
We are `using Plots` in the elixirs, so that's fine. We have 
```julia
using Test
using Trixi
```
at the top of most test files, so we need to replicate that since they're no longer available in the submodule created for each test case.

Reply:
Ah, ok. Thanks!